### PR TITLE
tools: Fix partial and empty matchers in rewrite

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: yeya24 <yb532204897@gmail.com>

Follow-up of #3886.

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Require full matchers matching when processing deletion requests. Only partial match or empty label is not allowed.

Let's say a deletion request is `{a="1", b="2"}`, so the series we want to delete must at least matches `{a="1", b="2"}`. It can contain other unrelated labels and it doesn't matter.

We don't want to delete series that have an empty label value of the matchers like `{c="1"}` or series that only has partial matches like `{a="1"}` or `{b="2"}`.

### CODE REVIEW

File: pkg/compactv2/modifiers.go [hunk 1]
```diff
@@ -62,12 +62,9 @@ SeriesLoop:
 		for _, deletions := range d.d.deletions {
 			for _, m := range deletions.Matchers {
 				v := lbls.Get(m.Name)
-				if v == "" {
-					continue
-				}
 
 				// Only if all matchers in the deletion request are matched can we proceed to deletion.
-				if !m.Matches(v) {
+				if v == "" || !m.Matches(v) {
```
Comment (yeya24) [inline pkg/compactv2/modifiers.go:67]: `lbls.Get(m.Name)` returns an empty string if the label is not present in the label set. What if the label name is present but with an empty string as its value, is this a valid case in Prometheus?

If yes, we need to use `lbls.Has(m.Name)` here.

Comment (squat) [reply pkg/compactv2/modifiers.go]: ack, this is a good point, @yeya24

**Review by squat — APPROVED** [review]

File: pkg/compactv2/modifiers.go (hunk 1 above)
Comment (yeya24) [reply pkg/compactv2/modifiers.go:67]: Just checked Prometheus and it is not allowed. So checking `v == ""` is fine.

Comment (squat) [reply pkg/compactv2/modifiers.go:67]: that's right, having empty value is the same as not having the label at all in Prometheus

**Review by squat — APPROVED** [review]

**Pull request merged**